src/main/java/org/springframework/data/elasticsearch/core/RequestFactory.java

```diff
@@ -888,6 +892,10 @@ public IndexRequest indexRequest(IndexQuery query, IndexCoordinates index) {
 			indexRequest.setIfPrimaryTerm(query.getPrimaryTerm());
 		}
 
+		if (query.getRouting() != null) {
```

Comment:
this needs to be added to `deleteByQueryRequestBuilder()` as well.

Reply:
Hi @sothawo , i think `routing` is already present in `updateRequest`
https://github.com/spring-projects/spring-data-elasticsearch/blob/cee03ee3b95ebae6dad9bff49322131e339ecaf3/src/main/java/org/springframework/data/elasticsearch/core/RequestFactory.java#L1361-#L1363
& for `updateRequestBuilder`
https://github.com/spring-projects/spring-data-elasticsearch/blob/cee03ee3b95ebae6dad9bff49322131e339ecaf3/src/main/java/org/springframework/data/elasticsearch/core/RequestFactory.java#L1435-#L1437